### ISSUE TITLE
kubectl describe shows SandboxChanged which is not a helpful user message

### ISSUE THREAD

**Issue by krmayankk** [kubernetes/kubernetes#62504]
I see the following kubectl describe message on a pod 

```
 Normal   SandboxChanged         3m (x8 over 3m)  kubelet, hostname.net  Pod sandbox changed, it will be killed and re-created.
```

Further digging into the kubelet logs, i see the following
```
Error response from daemon: {"message":"no available IPv4 addresses on this network's address pools: bridge (872bba0c06bf6f65e6185240b5b6
```
It will be so much more awesome if we can actually display this helpful message in the events for that pod.  SandBoxChanged to a cluster administrator has no meaning

```
kubectl version
Client Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.14", GitCommit:"d1303b003001cf2b5b8cec099383125096b3dac0", GitTreeState:"clean", BuildDate:"2018-03-12T16:24:22Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.14", GitCommit:"d1303b003001cf2b5b8cec099383125096b3dac0", GitTreeState:"clean", BuildDate:"2018-03-12T16:16:03Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by Davidrjx** [comment]
@krmayankk, in which tool did you deploy your cluster? your problem also seems much like network problem and you may be have a try to change cluster net solution, like version downgrade or the other net plugin.

**Comment by krmayankk** [comment]
@Davidrjx i am not concerned about the specific problem in hand. We know why we are seeing the network issue which is basically because we have only limited ip per host. What i am  suggesting here if instead of displaying `SandboxChanged` event for the pod, we can actually show the real error coming out from the docker which would be super helpful. SandboxChanged doesnt mean anything to the cluster operator

**Comment by jayunit100** [comment]
iirc sandbox changed means the config map or some other part of the pods setup changed?  should we just say that "the pod configuration inputs changed" ?

**Comment by ReDim42** [comment]
Maybe see #56996

**Comment by rbtcollins** [comment]
Seems to be that this is a legitimate message that is just a diagnostic for why a pod is restarted. But, when something causes a persistent error (e.g. failing to get an IP as the OP reported), then the problem is that this is the only event being reported, rather than being additional information.

So - I'm not sure this bug is valid *per se*, but whatever is causing these other errors to not be reported as events should be fixed and perhaps this bug is a decent placeholder for that work?

**Comment by mattparkes** [comment]
We just hit this case, which makes it hard to diagnose what the _actual_ problem with the Pod/Container was.

Can this ticket be re-opened?

**Comment by udipl** [comment]
We also just hit the same problem. There can be many diverse root causes for a `SandboxChanged` event, and cluster administrators could use information about the root cause here.

**Comment by inderps** [comment]
We also have the same problem. Would be awesome if somebody can tell how to debug further?

<img width="1155" alt="Screen Shot 2019-07-30 at 18 22 06" src="https://user-images.githubusercontent.com/1210046/62147037-f5f9d280-b2f6-11e9-856b-faa3dc26abc2.png">

**Comment by udipl** [comment]
In our case, we reviewed the kubelet logs on the node, which indicated that the pod's entrypoint process was failing. Then we had to look more broadly at **all** the system logs on the node to diagnose that the OS was killing the process because of memory pressure. It turned out that our RAM resource limits were set too low.
It would have been very helpful if that root cause would have been floated up to the cluster events.
